### ISSUE TITLE
Cannot read property 'parentNode' of null at ResponsiveComponent.handleWindowResize (fitWidth.js:104)

### ISSUE THREAD

**Issue by sulliwane** [rrag/react-stockcharts#247]
Hello,

I get this error when resizing the chrome window:
```bash
Uncaught TypeError: Cannot read property 'parentNode' of null at ResponsiveComponent.handleWindowResize (fitWidth.js:104)
handleWindowResize @ fitWidth.js:104
```
My CandlestickChart component is:
```js
const CandlestickChart = ({ data, ratio, width, height }) => {
  if (data.length > 0) {
    const startDate = data[0].date;
    const endDate = data[data.length - 1].date;
    return (
        <ChartCanvas ratio={ratio} width={width-20} height={height}
					className="CandlestickChart"
          margin={{ left: 50, right: 70, top: 10, bottom: 30 }}
          seriesName="MSFT"
          data={data} type={'hybrid'}
          xAccessor={d => d.date}
          xScaleProvider={discontinuousTimeScaleProvider}
          xExtents={[startDate, endDate]}
          clamp={true}
        >
          <Chart id={0} yExtents={d => [d.high, d.low]} height={400}>
            <XAxis axisAt="bottom" orient="bottom" stroke="#FFFFFF" tickStroke="#FFFFFF" />
            <YAxis axisAt="left" orient="left" ticks={5} stroke="#FFFFFF" tickStroke="#FFFFFF" />

            <CandlestickSeries
                  wickStroke={d => d.close > d.open ? "#DB0000" : "#6BA583"}
                  fill={d => d.close > d.open ? "#DB0000" : "#6BA583"}
            />

            <EdgeIndicator itemType="last" orient="right" edgeAt="right"
                  yAccessor={d => d.close} fill={d => d.close > d.open ? "#DB0000" : "#6BA583" }/>

            <MouseCoordinateX
                  at="bottom"
                  orient="bottom"
                  displayFormat={timeFormat("%Y-%m-%d")}
            />
            <MouseCoordinateY
                  at="right"
                  orient="right"
                  displayFormat={format(".2f")}
            />
            <OHLCTooltip origin={[10, 0]}/>
          </Chart>
            <Chart id={1} origin={(w, h) => [0, h - 150]} height={150} yExtents={d => d.volume}>
            <XAxis axisAt="bottom" orient="bottom" stroke="#FFFFFF" tickStroke="#FFFFFF"/>
            <YAxis axisAt="left" orient="left" ticks={5} tickFormat={format(".0s")} stroke="#FFFFFF"             tickStroke="#FFFFFF"/>
            <BarSeries yAccessor={d => d.volume} fill={(d) => d.close > d.open ? "#DB0000" : "#6BA583"} />
            </Chart>
            <CrossHairCursor stroke="#FFFFFF" />
            </ChartCanvas>
      )
  }
  return (
    <div className="spinner-box" style={{ height, width }}>
      <div className="sk-spinner-pulse"></div>
    </div>
  )
}

CandlestickChart.defaultProps = {
	height: 600,
};

export default fitWidth(CandlestickChart);
```
And the container div is:
```html
<div class="Content-box">
    <div class="react-stockchart" style="position: relative; width: 1025px; height: 400px;">
    </div>
</div>
```
Styles being:
```css
.Content-box {
    padding: 3px;
    color: white;
    background-color: #282c34;
    display: flex;
    flex-grow: 1;
    overflow-y: auto;
    height: 0;
    min-width: 100%;
}
```
So I suppose ReactSotckcharts is not finding the surrounding div (Content-box)? Any idea what could cause this problem?

Many thanks!

btw, this is a really awsome react project, I love it :+1:

**Comment by rrag** [comment]
Could you change your `CandlestickChart` from a stateless function component to a `class CandlestickChart extends React.Component`

**Comment by sulliwane** [comment]
thank you for your reply. Still same problem after changing to:

```js
class CandlestickChart extends Component {
  render() {
    const { data, ratio, width, height } = this.props
    if (data.length > 0) {
      const startDate = data[0].date;
      const endDate = data[data.length - 1].date;
      return (
          <ChartCanvas ratio={ratio} width={width-20} height={height}
  					className="CandlestickChart"
            margin={{ left: 50, right: 70, top: 10, bottom: 30 }}
            seriesName="MSFT"
            data={data} type={'hybrid'}
            xAccessor={d => d.date}
            xScaleProvider={discontinuousTimeScaleProvider}
            xExtents={[startDate, endDate]}
            clamp={true}
          >
            <Chart id={0} yExtents={d => [d.high, d.low]} height={400}>
              <XAxis axisAt="bottom" orient="bottom" stroke="#FFFFFF" tickStroke="#FFFFFF" />
              <YAxis axisAt="left" orient="left" ticks={5} stroke="#FFFFFF" tickStroke="#FFFFFF" />

              <CandlestickSeries
  							wickStroke={d => d.close > d.open ? "#DB0000" : "#6BA583"}
  							fill={d => d.close > d.open ? "#DB0000" : "#6BA583"}
  						/>

  						<EdgeIndicator itemType="last" orient="right" edgeAt="right"
  							yAccessor={d => d.close} fill={d => d.close > d.open ? "#DB0000" : "#6BA583" }/>

  						<MouseCoordinateX
  							at="bottom"
  							orient="bottom"
  							displayFormat={timeFormat("%Y-%m-%d")}
  						/>
  						<MouseCoordinateY
  							at="right"
  							orient="right"
  							displayFormat={format(".2f")}
  						/>
  						<OHLCTooltip origin={[10, 0]}/>
            </Chart>
  					<Chart id={1} origin={(w, h) => [0, h - 150]} height={150} yExtents={d => d.volume}>
  						<XAxis axisAt="bottom" orient="bottom" stroke="#FFFFFF" tickStroke="#FFFFFF"/>
  						<YAxis axisAt="left" orient="left" ticks={5} tickFormat={format(".0s")} stroke="#FFFFFF" tickStroke="#FFFFFF"/>
  						<BarSeries yAccessor={d => d.volume} fill={(d) => d.close > d.open ? "#DB0000" : "#6BA583"} />
  					</Chart>
  					<CrossHairCursor stroke="#FFFFFF" />
          </ChartCanvas>
        )
    }
    return (
      <div className="spinner-box" style={{ height, width }}>
        <div className="sk-spinner-pulse"></div>
      </div>
    )
  }
}
```

**Comment by rrag** [comment]
I am not sure, it is hard to tell without a plunker

one more thought, load the data in something outside, and only render this when you have the `data`

do not have conditional return inside like
```
    if (data.length > 0) {
```

**Comment by sulliwane** [comment]
Sorry for the late reply, I was away from the keyboard these days.

I just tried your second suggestion with no chance. But to be sure, I also tried again to replace the stateless component to a React Component child, and indeed, it worked !! 

My bad, I was in a hurry last time I tried, and I must have did something wrong in my test. 

So to recap, this error had nothing to do with the conditional rendering, but **the problem was due to my component not inheriting React Component class**.

Thanks again for your help!
